ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/queries/CommonQueries.scala

```diff
@@ -58,16 +55,21 @@ trait CommonQueries extends Queries {
 
   override final def selectStateValuesByKeys(keys: Seq[Key]): Try[immutable.Seq[Option[Value]]] =
     Try {
-      val results = SQL"SELECT key, value FROM #$StateTable WHERE key IN ($keys)"
-        .fold(Map.newBuilder[ByteString, Array[Byte]], ColumnAliaser.empty)((builder, row) =>
-          builder += ByteString.readFrom(row[InputStream]("key")) -> row[Value]("value"))
-        .fold(exceptions => throw exceptions.head, _.result())
-      keys.map(key => results.get(ByteString.copyFrom(key)))(breakOut)
+      val results =
+        SQL"SELECT key, value FROM #$StateTable WHERE key IN ($keys)"
+          .fold(Map.newBuilder[Key, Value], ColumnAliaser.empty) { (builder, row) =>
+            val key = row("key")
+            val value = row("value")
```

Comment:
We can probably inline these again; I only pulled them out because they were really long.

Reply:
Done.